preset/k3s/preset_test.go

```diff
@@ -122,3 +123,77 @@ func TestConfigBytes(t *testing.T) {
 		require.Contains(t, err.Error(), "connection refused")
 	})
 }
+
+func TestPreset_WithDynamicPort(t *testing.T) {
+	t.Parallel()
+
+	p := k3s.Preset(
+		k3s.WithDynamicPort(),
+	)
+	c, err := gnomock.Start(p, gnomock.WithContainerName("k3s-dynamic"))
+	require.NoError(t, err)
+
+	defer func() {
+		require.NoError(t, gnomock.Stop(c))
+	}()
+
+	kubeconfig, err := k3s.Config(c)
+	require.NoError(t, err)
+
+	client, err := kubernetes.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+
+	pods, err := client.CoreV1().Pods(metav1.NamespaceDefault).List(ctx, metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Empty(t, pods.Items)
+}
+
+func TestPreset_Versions(t *testing.T) {
+	tests := []struct {
+		inVersion string
+	}{
+		{"v1.16.7"},
+		{"v1.17.13"},
+		{"v1.18.10"},
+		{"v1.19.12"},
+		{"v1.20.8"},
+		{"v1.21.2"},
+		{"v1.22.17-k3s1"},
+		{"v1.23.17-k3s1"},
+		{"v1.24.12-k3s1"},
+		{"v1.25.8-k3s1"},
+		{"v1.26.3-k3s1"},
```

Comment:
Testing all these versions takes too much time in CI. What do you think about limiting the tests to whatever we claim to support in the readme?

Reply:
Perfect, that's helped. Just supporting the 4 released in a year doesn't time out anymore